src/connectors/range-slider/connectRangeSlider.js

```diff
@@ -76,108 +80,146 @@ export default function connectRangeSlider(renderFn) {
     };
 
     return {
-      getConfiguration: originalConf => {
-        const conf = {
-          disjunctiveFacets: [attributeName],
-        };
+      _isAbleToRefine(min, max) {
+        const isMinValid = _isFinite(min);
+        const isMaxValid = _isFinite(max);
 
-        const hasUserBounds = userMin !== undefined || userMax !== undefined;
-        const boundsNotAlreadyDefined =
-          !originalConf ||
-          (originalConf.numericRefinements &&
-            originalConf.numericRefinements[attributeName] === undefined);
-
-        if (hasUserBounds && boundsNotAlreadyDefined) {
-          conf.numericRefinements = { [attributeName]: {} };
-          if (userMin !== undefined)
-            conf.numericRefinements[attributeName]['>='] = [userMin];
-          if (userMax !== undefined)
-            conf.numericRefinements[attributeName]['<='] = [userMax];
+        if (isMinValid && isMaxValid && min >= max) {
+          return false;
         }
 
-        return conf;
+        return isMinValid || isMaxValid;
       },
 
-      _getCurrentRefinement(helper) {
-        let min = helper.state.getNumericRefinement(attributeName, '>=');
-        let max = helper.state.getNumericRefinement(attributeName, '<=');
-
-        if (min && min.length) {
-          min = min[0];
+      _getCurrentRange(stats = {}) {
+        let min;
+        if (isMinBounds) {
+          min = minBounds;
+        } else if (_isFinite(stats.min)) {
+          min = stats.min;
         } else {
-          min = -Infinity;
+          min = 0;
```

Comment:
I don't get why 0 could be a good value here. Can you explain?

Reply:
I tried to kept the compatibility.

In the previous version the range was compute with the stats object in the [init](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/range-slider/connectRangeSlider.js#L169) and the [render](https://github.com/algolia/instantsearch.js/blob/develop/src/connectors/range-slider/connectRangeSlider.js#L194) functions. Then thoses values are rounded when passed to the rendering function. But it might happen that the stats object can be `{ min: null, max: null }` and `null` is rounded to `0`.

The `0` case is used in `RangeSlider` for [hide it](https://github.com/algolia/instantsearch.js/blob/develop/src/widgets/range-slider/range-slider.js#L44) when there are nothing to display. In upcoming version we can imagine to pass a `disabled` property instead of rely on range values.